src/qibolab/serialize.py

```diff
@@ -120,7 +120,9 @@ def dump_qubits(qubits: QubitMap, pairs: QubitPairMap, couplers: CouplerMap = No
 def dump_instruments(instruments: InstrumentMap) -> dict:
     """Dump instrument settings to a dictionary following the runcard format."""
     return {
-        name: asdict(instrument.settings) for name, instrument in instruments.items() if instrument.settings is not None
+        name: asdict(instrument.settings, dict_factory=instrument.settings.dict_factory)
```

Comment:
Actually, maybe it would be even more intuitive if this were implemented as
```suggestion
        name: instrument.settings.dump()
```
and you delegate the usage of `dict_factory()` to the `dump()` method.